### ISSUE TITLE
7.0 Beta1 - GetAliasAsync returns incorrect result for IsValid when alias is missing.

### ISSUE THREAD

**Issue by niemyjski** [elastic/elasticsearch-net#3828]
We have the following Foundatio test at a high level

```cs
        [Fact]
        public async Task CanCreateAndDeleteVersionedIndex() {
            var index = new VersionedEmployeeIndex(_configuration, 1);

            await index.ConfigureAsync();
            var existsResponse = await _client.Indices.ExistsAsync(index.VersionedName);
            _logger.LogTraceRequest(existsResponse);
            Assert.True(existsResponse.IsValid);
            Assert.True(existsResponse.Exists);

            await _client.AssertSingleIndexAlias(index.VersionedName, index.Name);

            await index.DeleteAsync();
            existsResponse = await _client.Indices.ExistsAsync(index.VersionedName);
            _logger.LogTraceRequest(existsResponse);
            Assert.True(existsResponse.IsValid);
            Assert.False(existsResponse.Exists);

            Assert.Equal(0, await _client.GetAliasIndexCount(index.Name));
        }

42:24.77926 I:VersionedEmployeeIndex - 200 PUT http://elastic.localtest.me:9200/employees-v1?pretty=true
{"aliases":{"employees":{}},"mappings":{"properties":{"id":{"type":"keyword"},"companyId":{"fields":{"keyword":{"ignore_above":256,"type":"keyword"}},"type":"text"},"companyName":{"fields":{"keyword":{"ignore_above":256,"type":"keyword"}},"type":"text"},"name":{"fields":{"keyword":{"ignore_above":256,"type":"keyword"}},"type":"text"},"age":{"type":"integer"},"location":{"fields":{"keyword":{"ignore_above":256,"type":"keyword"}},"type":"text"},"yearsEmployed":{"type":"integer"},"lastReview":{"type":"date"},"nextReview":{"type":"date"},"createdUtc":{"type":"date"},"updatedUtc":{"type":"date"},"version":{"fields":{"keyword":{"ignore_above":256,"type":"keyword"}},"type":"text"},"isDeleted":{"type":"boolean"},"peerReviews":{"properties":{"reviewerEmployeeId":{"fields":{"keyword":{"ignore_above":256,"type":"keyword"}},"type":"text"},"rating":{"type":"integer"}},"type":"object"},"data":{"properties":{"comparer":{"properties":{},"type":"object"},"count":{"type":"integer"},"keys":{"properties":{"count":{"type":"integer"}},"type":"object"},"values":{"properties":{"count":{"type":"integer"}},"type":"object"},"item":{"properties":{},"type":"object"}},"type":"object"},"phoneNumbers":{"properties":{"number":{"fields":{"keyword":{"ignore_above":256,"type":"keyword"}},"type":"text"},"extension":{"fields":{"keyword":{"ignore_above":256,"type":"keyword"}},"type":"text"}},"type":"object"},"deleted":{"path":"isDeleted","type":"alias"},"created":{"path":"createdUtc","type":"alias"},"updated":{"path":"updatedUtc","type":"alias"}}},"settings":{"index.number_of_replicas":0,"index.number_of_shards":1}}

42:24.79834 T:IndexTests - [200] HEAD /employees-v1?pretty=true
42:24.81412 T:VersionedEmployeeIndex - [200] GET /employees/_alias?pretty=true
42:24.85080 T:VersionedEmployeeIndex - 200 DELETE http://elastic.localtest.me:9200/employees-v1%2Cemployees-v1-error?pretty=true&ignore_unavailable=true
{}
```

```cs
        public static async Task<int> GetAliasIndexCount(this IElasticClient client, string aliasName) {
            var response = await client.Indices.GetAliasAsync(aliasName, a => a.IgnoreUnavailable());
            if (!response.IsValid && response.ServerError?.Status == 404)
                return 0;
            
            Assert.True(response.IsValid);
            return response.Indices.Count;
        }
```

The last call to GetAliasIndexCount calls `Invalid NEST response built from a successful low level call on GET: /employees/_alias?pretty=true&ignore_unavailable=true` I called that http end point and it returned `{ }`

It should not be returning IsValid = false in this case.

**Comment by codebrain** [comment]
Relates to: https://github.com/elastic/elasticsearch-net/issues/3707

**Comment by Mpdreamz** [comment]
This was changed because as of 5.5 the following:

```json

PUT /x

PUT /x/_alias/z

GET /x/_alias/y,z
```

Actually returns a 404

```json
{
  "error" : "alias [y] missing",
  "status" : 404,
  "x" : {
    "aliases" : {
      "z" : { }
    }
  }
}

```

Eventhough `z` is returned. Therefor we updated the `IsValid` property to 

```csharp
public override bool IsValid => Indices.Count > 0;
```
This is good for the `404` partially being valid case but if the server response with 200 we don't need the extra validation.

 Thanks for raising this @niemyjski !

**Comment by Mpdreamz** [comment]
This should be fixed once #3873 lands

**Comment by codebrain** [comment]
Closing as #3873 has been merged, this should therefore be fixed in the next 7.0.0 release.
